src/manage.c

```diff
@@ -2977,26 +2977,46 @@ slave_setup (gvm_connection_t *connection, const char *name, task_t task,
               privacy_algorithm
                 = credential_iterator_privacy_algorithm (&credentials);
 
-              if (community == NULL || user == NULL || password == NULL
-                  || auth_algorithm == NULL || privacy_password == NULL
-                  || privacy_algorithm == NULL)
+              if (password && strcmp (password, "")
+                  && (auth_algorithm == NULL || strcmp (auth_algorithm, "")))
                 {
                   cleanup_iterator (&credentials);
                   global_slave_esxi_credential_uuid = NULL;
                   g_warning ("Could not create slave SNMP credential"
-                             " (missing community, login, password,"
-                             " auth_algorithm, privacy_password"
-                             " or privacy_algorithm)."
+                             " (auth_algorithm must be set if password is"
+                             " given)."
+                             " Continuing without credential.");
+                }
+              if (((privacy_password && strcmp (privacy_password, ""))
```

Comment:
Surely this should be an "else if", because the if above says it will fail if true.

Reply:
You're right. I've changed it in d01d4f3 as well.